samples/nrf9160/gps/README.rst

```diff
@@ -1,19 +1,26 @@
 .. _gps_with_supl_support_sample:
 
-nRF9160: GPS sockets and SUPL client library
-############################################
+nRF9160: GNSS interface and SUPL client library
+###############################################
 
 .. contents::
    :local:
    :depth: 2
 
-The GPS sample demonstrates how to retrieve `GPS`_ data.
-If Secure User-Plane Location (SUPL) support is enabled, it also shows how to improve the GPS fix accuracy and fix speed with `A-GPS`_ data from a SUPL server.
+The GPS sample demonstrates how to retrieve `GNSS`_ data.
+If Secure User-Plane Location (SUPL) support is enabled, it also shows how to improve the fix accuracy and fix speed with `A-GPS`_ data from a SUPL server.
 See :ref:`supl_client` for information on enabling SUPL support for the sample.
 
-The sample first creates a `GNSS`_ socket.
-Then it reads the GPS data from the socket and parses the received frames to interpret the frame contents.
-See :ref:`nrfxlib:gnss_extension` for more information.
+The sample first initializes the `GNSS`_ interface.
+Then it handles events from the interface, reads the associated data and outputs information to the console.
+Because `NMEA`_ data needs to be read as soon as an `NMEA`_ event is received, a message queue is used for buffering the `NMEA`_ strings.
+The event handler function reads the received `NMEA`_ strings and puts those into the queue.
+The consumer loop reads from the queue and outputs the strings to the console.
+
+When SUPL support is enabled, a workqueue is used for downloading the `A-GPS`_ data.
```

Comment:
```suggestion
When SUPL support is enabled, a work queue is used for downloading the `A-GPS`_ data.
```

Reply:
I'm referring to Zephyr workqueues, which are written without a space: https://docs.zephyrproject.org/1.9.0/kernel/threads/workqueues.html

Should we add a link to the Zephyr documentation?

Reply:
Could be a good idea. Generally, work queue seems to be written as two separate words as I suggested, but Zephyr seems to disagree with that.